build.go

```diff
@@ -4,42 +4,47 @@ import (
 	"fmt"
 	"path/filepath"
 
-	"github.com/paketo-buildpacks/packit"
-	"github.com/paketo-buildpacks/packit/scribe"
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
 )
 
 //go:generate faux --interface Parser --output fakes/parser.go
 type Parser interface {
 	Parse(path string) (hasRakeGem bool, err error)
 }
 
-func Build(gemfileParser Parser, logger scribe.Logger) packit.BuildFunc {
+func Build(gemfileParser Parser, logger scribe.Emitter) packit.BuildFunc {
 	return func(context packit.BuildContext) (packit.BuildResult, error) {
 		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
 
+		logger.Debug.Process("Checking Gemfile for rake")
 		hasRakeGem, err := gemfileParser.Parse(filepath.Join(context.WorkingDir, "Gemfile"))
 		if err != nil {
 			return packit.BuildResult{}, fmt.Errorf("failed to parse Gemfile: %w", err)
 		}
 
-		command := "rake"
+		args := "rake"
 
 		if hasRakeGem {
-			command = "bundle exec rake"
+			logger.Debug.Subprocess("Gemfile contains rake gem")
+			args = "bundle exec rake"
 		}
-
-		logger.Process("Assigning launch processes")
-		logger.Subprocess("web: %s", command)
+		logger.Debug.Break()
+
+		processes := []packit.Process{
+			{
+				Type:    "web",
+				Command: "bash",
+				Args:    []string{"-c", args},
```

Comment:
I don't think we'd need to use `bash` as the command here because we don't have any env vars to interpolate in the arguments.